js/lib/beautify.js

```diff
@@ -1014,6 +1015,62 @@ function Beautifier(js_source_text, options) {
                 !(check_token.type === 'TK_END_BLOCK' && check_token.opened === current_token));
         }
 
+        if (flags.inline_frame === false && opt.align_object_literals) {
+            var setKeyPadding = null;
+            var maxKeyLen = 0;
+            var index, lookingAt, check_token, ignoreUntil;
+            var lastKeyLen = 0;
+            while (true) {
+                index = 0;
+                lookingAt = 'KEY';
+                check_token = null;
+                ignoreUntil = null;
+                do {
+                    index += 1;
+                    check_token = get_token(index);
+                    if ((lookingAt === 'VALUE' || lookingAt === 'VALUESTART') && ignoreUntil !== null) {
+                        if ((check_token.type === 'TK_END_BLOCK' || check_token.type === 'TK_END_EXPR') && check_token.opened === ignoreUntil) {
+                            ignoreUntil = null;
+                        }
+                        continue;
+                    }
+                    if (lookingAt === 'OPERATOR' && check_token.type == 'TK_OPERATOR') {
+                        lookingAt = 'VALUESTART';
+                        continue;
+                    }
+                    if (lookingAt === 'VALUESTART' || lookingAt === 'VALUE') {
+                        if (lookingAt === 'VALUESTART' && setKeyPadding !== null) {
+                            check_token.text = new Array(setKeyPadding - lastKeyLen + 1).join(' ') + check_token.text;
```

Comment:
this is my main dislike.  I tried setting whitespace_before to a string of whitespace, but that didn't appear to affect the output.  Is there a better way?